backend/substrapp/tasks/tasks.py

```diff
@@ -625,6 +626,33 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
             client.volumes.create(name=volume_id)
         model_volume[volume_id] = {'bind': '/sandbox/local', 'mode': 'rw'}
 
+    if os.getenv('CHAINKEYS_FOR_TRAINING_TASKS_ENABLED', False):
+        label_selector = f"compute_plan={subtuple['tag']}"
+        secrets = []
+
+        k8s_client = get_k8s_client()
+        try:
+            secret_namespace = os.getenv('K8S_SECRET_NAMESPACE', 'default')
+            secrets = k8s_client.list_namespaced_secret(secret_namespace, label_selector=label_selector)
+        except ApiException as e:
+            logging.error(f'failed to fetch namespaced secrets {secret_namespace} with selector {label_selector}')
+            raise e
+
+        chainkeys_directory = get_chainkeys_directory()
+        if not os.path.exists(chainkeys_directory):
+            os.mkdir(chainkeys_directory)
+            with open(path.join(chainkeys_directory, 'chainkeys.json'), 'w') as file:
+                file.write(str(secrets))
+
+        volumes[chainkeys_directory] = {'bind': '/sandbox/chainkeys', 'mode': 'rw'}
+
+    # Environment current node index
+    node_index = os.getenv('NODE_INDEX', None)
+    if node_index:
+        environment = {
```

Comment:
`environment` is declared as a `list` object line 601. It looks confusing.

What about creating an empty dict line 601 and updating it here?
```python
environment["NODE_INDEX"] = node_index
```